File: TreasurersApp/Models/EmailAddressActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class EmailAddressActionResult : WebResult<EmailAddress>
    {
        public EmailAddressActionResult(bool success, List<string> statusMessages, EmailAddress emailAddress)
            :base(success, statusMessages, emailAddress)
        {
        }
    }
}

File: TreasurersApp/ClientApp/src/app/models/EmailAddressActionResult.ts
import { EmailAddress } from './EmailAddress';
import { EmailService } from '../maintenance/email-maintenance/email.service';

export class EmailAddressActionResult {
  success = false;
  statusMessages: string[] = [];
  emailAddress: EmailAddress = null;

  constructor(private emailService: EmailService) {
    this.success = false;
    this.emailAddress = this.emailService.newEmailAddress();
    this.statusMessages = [];
  }
}
